Gems/ROS2/Code/Source/RobotImporter/URDF/URDFPrefabMaker.cpp

```diff
@@ -0,0 +1,332 @@
+        m_collidersMaker.ProcessMeshes(buildReadyCb);
+    }
+
+    void URDFPrefabMaker::BuildAssetsForLink(urdf::LinkSharedPtr link)
+    {
+        m_collidersMaker.BuildColliders(link);
+        for (auto childLink : link->child_links)
+        {
+            BuildAssetsForLink(childLink);
+        }
+    }
+
+    AzToolsFramework::Prefab::CreatePrefabResult URDFPrefabMaker::CreatePrefabFromURDF()
+    {
+        {
+            AZStd::lock_guard<AZStd::mutex> lck(m_statusLock);
+            m_status.clear();
+        }
+        // TODO - this is PoC code, restructure when developing semantics of URDF->Prefab/Entities/Components mapping
+        AZStd::unordered_map<AZStd::string, AzToolsFramework::Prefab::PrefabEntityResult> created_links;
+        AzToolsFramework::Prefab::PrefabEntityResult createEntityRoot = AddEntitiesForLink(m_model->root_link_, AZ::EntityId());
+        AZStd::string root_name(m_model->root_link_->name.c_str(), m_model->root_link_->name.size());
+        created_links[root_name] = createEntityRoot;
+        if (!createEntityRoot.IsSuccess())
+        {
+            return AZ::Failure(AZStd::string(createEntityRoot.GetError()));
+        }
+
+        auto links = Utils::GetAllLinks(m_model->root_link_->child_links);
+
+        // create links
+        for (const auto& [name, link_ptr] : links)
+        {
+            created_links[name] = AddEntitiesForLink(link_ptr, createEntityRoot.GetValue());
+        }
+
+        for (const auto& [name, result] : created_links)
+        {
+            AZ_TracePrintf(
+                "CreatePrefabFromURDF",
+                "Link with name %s was created as: %s",
+                name.c_str(),
+                result.IsSuccess() ? (result.GetValue().ToString().c_str()) : ("[Failed]"));
+            AZStd::lock_guard<AZStd::mutex> lck(m_statusLock);
+            if (result.IsSuccess())
+            {
+                m_status.emplace(name, AZStd::string::format("created as: %s", result.GetValue().ToString().c_str()));
+            }
+            else
+            {
+                m_status.emplace(name, AZStd::string::format("failed : %s", result.GetError().c_str()));
+            }
+        }
+
+        // set transforms of links
+        for (const auto& [name, link_ptr] : links)
+        {
+            const auto this_entry = created_links.at(name);
+            if (this_entry.IsSuccess())
+            {
+                AZ::Transform tf = Utils::GetWorldTransformURDF(link_ptr);
+                auto* entity = AzToolsFramework::GetEntityById(this_entry.GetValue());
+                if (entity)
+                {
+                    auto* transformInterface = entity->FindComponent<AzToolsFramework::Components::TransformComponent>();
+                    if (transformInterface)
+                    {
+                        AZ_TracePrintf(
+                            "CreatePrefabFromURDF",
+                            "Setting transform %s %s to [%f %f %f] [%f %f %f %f]",
```

Comment:
This `AZ_TracePrintf` should have a newline at the end of the format string.
```suggestion
                            "Setting transform %s %s to [%f %f %f] [%f %f %f %f]\n",
```